### PR TITLE
Add OpenSSL build and tests

### PR DESCRIPTION
**Pull request by Blackhex**
Builds OpenSSL and executes its tests.

### CODE REVIEW

**Review by ZacWalk — APPROVED** [review]
LGTM

File: .github/scripts/build-openssl.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+OPENSSL_BUILD_PATH=$BUILD_PATH/openssl
+
+mkdir -p $OPENSSL_BUILD_PATH
+cd $OPENSSL_BUILD_PATH
+
+if [ $RUN_CONFIG = 1 ] || [ ! -f "$OPENSSL_BUILD_PATH/Makefile" ] ; then
+    echo "::group::Configure OpenSSL"
+        rm -rf $OPENSSL_BUILD_PATH/*
+        $SOURCE_PATH/$OPENSSL_VERSION/Configure \
```
Comment (eukarpov) [inline .github/scripts/build-openssl.sh:13]: Could you please check if it enables assembly optimization on aarch64?

**Review by eukarpov — APPROVED** [review]

File: .github/scripts/build-openssl.sh (hunk 1 above)
Comment (Blackhex) [reply .github/scripts/build-openssl.sh:13]: It does use `-O3`.

Comment (eukarpov) [reply .github/scripts/build-openssl.sh:13]: Does it compile aarch64 assembly to enable extra optimization on OpenSSL?

Comment (Blackhex) [reply .github/scripts/build-openssl.sh:13]: How will I find out?

Comment (eukarpov) [reply .github/scripts/build-openssl.sh:13]: I checked the logs it looks it is included

`2023-12-13T10:43:59.0030967Z CC="/home/runner/work/mingw-woarm64-build/mingw-woarm64-build/cross/bin/aarch64-w64-mingw32-gcc" /usr/bin/perl ../../code/openssl-master/crypto/sha/asm/sha512-armv8.pl "win64" -Icrypto -I../../code/openssl-master/crypto -I. -Iinclude -Iproviders/common/include -Iproviders/implementations/include -I../../code/openssl-master -I../../code/openssl-master/include -I../../code/openssl-master/providers/common/include -I../../code/openssl-master/providers/implementations/include -march=armv8-a+crypto -Wall -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/home/runner/work/mingw-woarm64-build/mingw-woarm64-build/openssl/ssl\"" -DENGINESDIR="\"/home/runner/work/mingw-woarm64-build/mingw-woarm64-build/openssl/lib64/engines-3\"" -DMODULESDIR="\"/home/runner/work/mingw-woarm64-build/mingw-woarm64-build/openssl/lib64/ossl-modules\"" -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -DBSAES_ASM -DECP_NISTZ256_ASM -DECP_SM2P256_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_MONT -DOPENSSL_CPUID_OBJ -DOPENSSL_SM3_ASM -DPOLY1305_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DSM4_ASM -DVPAES_ASM -DVPSM4_ASM  crypto/sha/sha256-armv8.S`

**Pull request merged**